scapy/layers/dot11.py

```diff
@@ -384,7 +384,7 @@ class Dot11Beacon(Packet):
 class Dot11Elt(Packet):
     name = "802.11 Information Element"
     fields_desc = [ByteEnumField("ID", 0, _dot11_info_elts_ids),
-                   FieldLenField("len", None, "info", "B"),
+                   ByteField("len", None),
```

Comment:
Dot11EltRSN and Dot11EltRates have no « info » parameter.
I replaced the use of info to get the length, by the use of `post_build`

Reply:
I see. However I think when possible we should use a length field (optionally with an `adjust=` parameter) is better as it is easy to understand what happens reading the output of `ls()` or the content of the `fields_desc` attribute.

Since you have to redefine `fields_desc` for `Dot11EltRSN` and `Dot11EltRates` anyway, you would just have to define the `len` field differently.

BTW, length fields are used for both building and dissecting.

As a side note, don't you think the implementation of specific `Dot11Elt` could be done more easily, now we have `MultipleTypeField`?

Reply:
When it's only a few different fields, I think `MultipleTypeField` is perfect. However, here Dot11Elt are completely different, so it will be unreadable